unittests/baseplatformtests.py

```diff
@@ -255,7 +255,7 @@ def init(self, srcdir, *,
                     self._print_meson_log()
                     raise
                 out = self._get_meson_log()  # best we can do here
-        return out
+        return out or ''
```

Comment:
This seems random, what's it trying to do?

Reply:
`get_meson_log()` can return `None`, but the function must return a `str`. If an exception occurs and there is no log to read, the function returns `None`, and it causes another exception when the test performs a test on the out value.